content/learn/book/control-flow/states.md

```diff
@@ -4,3 +4,118 @@ insert_anchor_links = "right"
 [extra]
 weight = 2
 +++
+
+Virtually all games are presented as a series of interactive "modes". Take for example a classic
+arcade game such as _Space Invaders_ or _Pac-Man_: First you have an "intro" mode, which catches the
+player's interest and teaches them about how the game works. The competition doesn't actually start until
+we enter the "play" mode, where the user is presented with various interactive challenges. At some
+point the game level will end, and the player will be presented with a "Congratulations, you Won!"
+or "Sorry, you Lost!" screen.
+
+Each of these modes entails a different set of graphics as well as a different configuration
+of input controls. During the intro mode, for example, all you can do is start the game, or perhaps
+get help or view the game's credits screen. You can't start controlling your ship or character
+until after you have left the "intro" mode and entered "play" mode.
+
+How shall we organize all of these different modal states in code? We could have separate variables for each
+of the modes, and write a bunch of logic for all the various entities involved, but there's an
+easier way: _game states_.
+
+Bevy's [`States`] trait lets you define one or more [finite-state
+machines](https://en.wikipedia.org/wiki/Finite-state_machine) (FSMs) that can be used to orchestrate
+changes to the Bevy world. You can configure your ECS systems to only run during certain states, or
+only on when transitioning between certain states. You can create entities that only exist
+within particular states as well. For example, you might have a HUD (heads-up display) entity which
+only exists during "play" mode.
+
+Let's say that we're building a retro arcade game like _Galaxian_. We'll want at least three states:
```

Comment:
No one is going to know what Galaxian is, sorry!

Reply:
WHAT.

Don't these young'uns remember anything?

Reply:
Galaga is a bit better :p Hard to remember things that I wasn't born for!

Reply:
I'll add a hyperlink for the chronologically challenged